### PR TITLE
[Flow] src/utils/editor/source-editor.js 

### PR DESCRIPTION
**Pull request by atwalg2**
Fixes Issue: #5120

### CODE REVIEW

File: src/utils/editor/source-editor.js [hunk 1]
```diff
@@ -70,15 +72,15 @@ export default class SourceEditor {
     }
   }
 
-  get codeMirror(): any {
+  get codeMirror(): CodeMirror {
```
Comment (AnshulMalik) [inline src/utils/editor/source-editor.js]: Umm, not sure if CodeMirror is a type defined somewhere. We probably need to get its types from somewhere.

Comment (darkwing) [reply src/utils/editor/source-editor.js]: IMO we should just use `object`; defining props for third party libraries seems unnecessary.

Comment (jasonLaster) [reply src/utils/editor/source-editor.js]: it is really nice to have 3rd party types because then we have better API docs :)

**Comment by atwalg2** [comment]
adding flow here causes some weird behavior in components/editor/index.

**Comment by jasonLaster** [comment]
@AnshulMalik @ryanjduffy thoughts?

**Comment by jasonLaster** [comment]
looks like getSource should be looser and handle nulls `?SourceId`
@AnshulMalik @ryanjduffy thoughts?

**Comment by AnshulMalik** [comment]
Yeah, we probably need to make it `?editor`

**Comment by atwalg2** [comment]
image

**Comment by AnshulMalik** [comment]
Well, this is strange

**Comment by darkwing** [comment]
I had a go at this, albeit it with a major headache, and ran into a hornets nest of issues.  Will look again tomorrow.

**Comment by atwalg2** [comment]
Awesome. Yea this is a weird bug I'm gonna go at it tonight and tomorrow as well

**Comment by ryanjduffy** [comment]
In this case, I think you're invalidating the type refinement. Flow doesn't know if `this.state.editor` is still valid after `resizeBreakpointGutter` since it _could_ have altered the object. The solution is to extract the `codeMirror` member after the check but before the function calls.

https://flow.org/en/docs/lang/refinements/#toc-refinement-invalidations

```js
const { codeMirror } = this.state.editor;

resizeBreakpointGutter(codeMirror);
resizeToggleButton(codeMirror);
```

**Comment by jasonLaster** [comment]
closing as this looks stuck. Feel free to re-open it if you figure it out...

**Pull request closed without merging**